### ISSUE TITLE
Move code-coverage activation to a Maven profile

### ISSUE THREAD

**Issue by eolivelli** [apache/bookkeeper#475]
We should move JaCoCo + Surefire @{argline} configuration in a separate profile. The risk is to alway run JaCoCo at every invocation if surefire.

**Comment by eolivelli** [comment]
We I committed the JaCoCo configuration I did not notice that now every execution of surefire uses JaCoCo I thought the activation of the instrumentation was only active when running the jacoco:report goal

**Comment by sijie** [comment]
@eolivelli I don't think it is needed to be a `blocker`. especially this is just a build problem. we should only mark issues that are really blocking a release.

**Comment by eolivelli** [comment]
@sijie I dropped the 'blocker' flag.
I added it because JaCoCo instruments bytecode and it can affect somehow the tests and I would not like that people run with JaCoCo every time they run a test.
In other projects I saw weird stacktraces in errors.
It also slow down the executions of tests.

Additional notes (not related to the issue):
I am also continuing to investigate if there is a way to build real aggregate reports, it seems that JaCoCo + Coveralls count as 'covered' only lines which are exercised in the same Maven project and non in dependant projects (like the stats providers/http server....)
